### ISSUE TITLE
Naming

### ISSUE THREAD

**Issue by adamabeshouse** [cBioPortal/oncoprintjs#9]
I think we may be having a naming issue - there are several things that just don't seem to have a natural name, and so we are ad-hoc making names for them in the code. I think it would be best if, at this point, we standardized terms for various elements of the oncoprint.

My main thrust here is to decouple visual concepts from biological concepts. 
The most obvious motivator for this is the mild confusion that arises when we have to deal with whether columns represent samples or patients, but we still refer to/think about each cell as a 'sample'. It would be better, I think, to just call them 'cells' and couple the concepts elsewhere in the code. 

As far as generality, if we want to add to the oncoprint later, it will be easier to make sense of what's going on if we don't have to reckon with using a visualization that's named after one biological concept to also refer to a similar yet technically distinct biological concept.

In short, this will make it easier to add to the oncoprint in the future, and will make it easier for us to mentally organize the code.

I've attached a Google slide and my proposals (based on what's in the code, and some ideas of my own). Go ahead and make changes/suggestions in the slides and let's figure it out together!

Some things you might do:
- challenge/endorse a name thats in there, or suggest a different name
- raise something that needs to be named that's not represented in the slide
- argue this isn't necessary :stuck_out_tongue: 

https://docs.google.com/presentation/d/1Q9m2SZfW2XIbX4PIi2d4BtbBD1En_nJHChWoOlw2fkE/edit?usp=sharing

**Comment by jjgao** [comment]
Great idea! I like your proposed names.

I think we should make the concepts as generic as possible. We probably don't event need any biological concepts (gene, clinical) in the abstract layer of implementation. We only need to attach the biological meaning to the data when defining the visualization them. 

Ideally, we implement a core module providing core functions such rendering, sorting, etc. We can attach visualization styles to the data by configuration. For genomics data, we then provide a preconfigured solution. Let's discuss when I am back on thursday.

**Comment by gideonite** [comment]
Great idea. I'll add this to the Trello board and to the documentation.

**Comment by gideonite** [comment]
I assume that this will be addressed by [PR 10](https://github.com/cBioPortal/oncoprintjs/pull/10).
